tools/azsdk-cli/docs/cli-commands-guidelines.md

```diff
@@ -0,0 +1,416 @@
+- `package test run --package-path ./sdk/keyvault --test-suite integration`
+- `package validate --package-path ./sdk/keyvault`
+
+### 3. **github** - GitHub Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.GitHub`  
+**Command Group:** Custom (no predefined group)  
+**Verb:** `github`
+
+For operations related to GitHub:
+- Create pull requests
+- Check PR status
+- Manage labels
+- Repository operations
+
+**Examples:**
+- `github create-pr --title "Update SDK" --description "Description"`
+- `github get-pr-details --pr 123`
+- `github labels sync`
+
+### 4. **release-plan** - Release Planning
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.ReleasePlan`  
+**Command Group:** Custom (no predefined group)  
+**Verb:** `release-plan` 
+
+For release planning and SDK coordination:
+- Create release plans
+- Update SDK details in release plans
+- Link SDK pull requests to release plans
+- Get release plan details
+- Generate SDK for all languages
+- Workflow management
+
+**Examples:**
+- `release-plan check-api-readiness --typespec-project ./spec/compute`
+- `release-plan generate-sdk --typespec-project ./spec/storage --api-version 2023-01-01`
+- `release-plan get --workitem-id 456`
+- `release-plan link-sdk-pr --release-plan-id 123 --pr 789`
+
+### 5. **typespec** - TypeSpec Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.TypeSpec`  
+**Command Group:** `SharedCommandGroups.TypeSpec`  
+**Verb:** `tsp`  
+
+For TypeSpec-related operations:
+- Create new TypeSpec projects
+- Convert Swagger to TypeSpec
+- Validate TypeSpec projects
+- Common TypeSpec workflows
+
+**Examples:**
+- `tsp convert --swagger-file ./swagger.json`
+- `tsp init --name MyService`
+- `tsp validate --project-path ./typespec`
+
+### 6. **tspclient** - TypeSpec Client Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.TypeSpec` (may have dedicated namespace)  
+**Command Group:** `SharedCommandGroups.TypeSpec` or custom  
+**Verb:** `tspclient` or integrated into `tsp`
+
+For TypeSpec client-specific operations:
+- Run TypeSpec client operations
+- Update TypeSpec client configurations
+- Client generation workflows
+
+**Examples:**
+- `tsp-client generate --output-dir ./generated`
+- `tsp-client update --config-path ./tspconfig.yaml`
+
+## Namespace Organization
+
+### Directory Structure
+```
+Tools/
+├── EngSys/           # Engineering systems commands
+├── Package/          # Package-level operations
+├── GitHub/           # GitHub-related operations  
+├── ReleasePlan/      # Release planning operations
+├── TypeSpec/         # TypeSpec operations
+└── Example/          # Example implementations (DEBUG only)
+```
+
+### Namespace Conventions
+- Use descriptive namespaces that match the directory structure
+- Follow the pattern: `Azure.Sdk.Tools.Cli.Tools.{Category}`
+- Each tool class should be in its appropriate namespace directory
+
+## Implementation Requirements
+
+### 1. Base Class Implementation
```

Comment:
I think this section is redundant with `new-tool.md`, or at least we should remove examples from that doc and have this be the canonical location.